// ng4-configure/src/ng4-configure.module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var http_1 = require("@angular/http");
var configure_options_1 = require("./configure-options");
var ng4_configure_service_1 = require("./ng4-configure.service");
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var NgConfigureModule = (function () {
    function NgConfigureModule() {
    }
    NgConfigureModule.forRoot = function () {
        return {
            ngModule: NgConfigureModule,
            providers: [ng4_configure_service_1.ConfigureService, configure_options_1.ConfigureOptions]
        };
    };
    NgConfigureModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [
                        common_1.CommonModule,
                        http_1.HttpModule
                    ],
                    declarations: [],
                    providers: [
                        ng4_configure_service_1.ConfigureService,
                        configure_options_1.ConfigureOptions,
                        {
                            provide: core_1.APP_INITIALIZER,
                            useFactory: init,
                            multi: true,
                            deps: [ng4_configure_service_1.ConfigureService, http_1.Http]
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    NgConfigureModule.ctorParameters = function () { return []; };
    return NgConfigureModule;
}());
exports.NgConfigureModule = NgConfigureModule;
function init(configService, http) {
    return function () { return configService.load(); };
}
exports.init = init;
